internal/provider/vaultsecrets/integration_utils.go

```diff
@@ -0,0 +1,109 @@
+package vaultsecrets
```

Comment:
I tried to group as much provider-agnostic logic as possible here so the integration resources for the next providers are easier to implement.